src/api/vdom.js

```diff
@@ -258,7 +267,7 @@ const newElementOpenEnd = wrapIdomFunc(elementOpenEnd);
 
 // Standard open / closed overrides don't need to reproduce internal behaviour
 // because they are the ones referenced from *End and *Start.
-const newElementOpen = wrapIdomFunc(elementOpen, stackOpen);
+const newElementOpen = attributesContext(wrapIdomFunc(elementOpen, stackOpen));
```

Comment:
Perhaps a useful upstream change to incremental-dom would be to allow the entire attributes object to be changed, rather than just attributes on it. This would allow context switches to be faster (single assignment operation) rather than being proportional to the number of attribute overrides.